### PR TITLE
Patch templatename in output for logic apps

### PR DESCRIPTION
**Pull request by fashaikh**
for logic apps templatename is not populated. the resourcegroup tag doesn't get set, or gets reset at a later point . Patching this for now. ACOM needs this to be populated to get the flow working

### CODE REVIEW

File: SimpleWAWS/ARMTemplates/BaseARMTemplate.json [hunk 1]
```diff
@@ -0,0 +1,103 @@
+﻿{
+  "$schema": "http://schemas.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "msdeployPackageUrl": {
+      "type": "string",
+      "defaultValue": "{{https://github.com/fashaikh/appservice-zipped-templates/raw/master/PHP/WordPress.zip}}"
```
Comment (fashaikh) [inline SimpleWAWS/ARMTemplates/BaseARMTemplate.json]: will move this to github.com/azure

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 2]
```diff
@@ -42,16 +45,17 @@ static TemplatesManager()
...
                             FileName = Path.GetFileName(template),
-                            Language = (language.Equals("Mobile", StringComparison.OrdinalIgnoreCase) || language.Equals("Api", StringComparison.OrdinalIgnoreCase))? null : language,
+                            Language = (language.Equals("Mobile", StringComparison.OrdinalIgnoreCase) || language.Equals("Api", StringComparison.OrdinalIgnoreCase)) ? null : language,
                             SpriteName = string.Format(CultureInfo.InvariantCulture, "{0} {1}", iconCssClass, cssClass),
-                            AppService =  language.Equals("Mobile", StringComparison.OrdinalIgnoreCase) ? AppService.Mobile : language.Equals("Api", StringComparison.OrdinalIgnoreCase)? AppService.Api : AppService.Web
+                            AppService = language.Equals("Mobile", StringComparison.OrdinalIgnoreCase) ? AppService.Mobile : language.Equals("Api", StringComparison.OrdinalIgnoreCase) ? AppService.Api : AppService.Web,
+                            MSDeployPackageUrl = $"https://github.com/fashaikh/appservice-zipped-templates/raw/master/{Path.GetDirectoryName(template)}/{Path.GetFileName(template)}"
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: Why not have these packages in our storage account instead? seems wrong to use github this way.

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 3]
```diff
@@ -60,35 +64,72 @@ static TemplatesManager()
...
+                    CsmTemplateFilePath = HostingEnvironment.MapPath("~/ARMTemplates/JenkinsResource.json"),
                     Description = Resources.Server.Templates_JenkinsDescription
                 });
+                _baseARMTemplate = JObject.Parse(File.ReadAllText(HostingEnvironment.MapPath("~/ARMTemplates/BaseARMTemplate.json"),  System.Text.Encoding.ASCII)
+                    .Replace(Environment.NewLine, String.Empty)
+                    .Replace(@"\", String.Empty));
                 //TODO: Implement a FileSystemWatcher for changes in the directory
             }
-            catch (Exception)
+            catch (Exception ex)
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: unused?

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 4]
```diff
@@ -60,35 +64,72 @@ static TemplatesManager()
...
             {
                 _templatesList = Enumerable.Empty<BaseTemplate>();
             }
         }
 
+        private static object getShortName(string templateName)
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: it's more of a css name or css sprite name than a general shortName. I think this method name is a bit confusing. Also it should be PascalCase

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 5]
```diff
@@ -60,35 +64,72 @@ static TemplatesManager()
...
+        {
+            return templateName.Replace(" ", "").Replace("#", "Sharp").Replace(".", "").Replace("+", "");
+        }
+
         public static IEnumerable<BaseTemplate> GetTemplates()
         {
             return _templatesList;
         }
+        public static JObject GetARMTemplate(BaseTemplate template)
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: nit: empty line before the function

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 6]
```diff
@@ -60,35 +64,72 @@ static TemplatesManager()
...
+        {
+            var armTemplate = _baseARMTemplate;
+            updateParameters(ref armTemplate, template);
+            //TODO: Add mroe specific app settings as needed.
+            updateAppSettings(ref armTemplate, template);
+            updateConfig(ref armTemplate, template);
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: why `ref`?

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 7]
```diff
@@ -60,35 +64,72 @@ static TemplatesManager()
...
+            return templateName.Replace(" ", "").Replace("#", "Sharp").Replace(".", "").Replace("+", "");
+        }
+
         public static IEnumerable<BaseTemplate> GetTemplates()
         {
             return _templatesList;
         }
+        public static JObject GetARMTemplate(BaseTemplate template)
+        {
+            var armTemplate = _baseARMTemplate;
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: This method is not thread safe. It changes a common global object then return it?

File: SimpleWAWS/Models/ResourceGroup.cs [hunk 8]
```diff
@@ -181,6 +181,8 @@ public UIResource UIResource
                         break;
                 }
                 var templateName = Tags.ContainsKey(Constants.TemplateName) ? Tags[Constants.TemplateName] : string.Empty;
+                if (String.Equals(templateName, string.Empty) && AppService == AppService.Logic)
```
Comment (ahmelsayed) [inline SimpleWAWS/Models/ResourceGroup.cs]: _baseARMTemplate

File: SimpleWAWS/Controllers/TemplatesController.cs [hunk 9]
```diff
@@ -39,5 +46,18 @@ public IEnumerable<BaseTemplate> Get()
             list.Add(WebsiteTemplate.EmptySiteTemplate);
             return list;
         }
+        public HttpResponseMessage GetARMTemplate(string templateName)
+        {
+            var list = TemplatesManager.GetTemplates().ToList();
```
Comment (ahmelsayed) [inline SimpleWAWS/Controllers/TemplatesController.cs]: You don't really need .ToList(). I thought GetTemplates() already return an IEnumerable

File: SimpleWAWS/Controllers/TemplatesController.cs [hunk 10]
```diff
@@ -39,5 +46,18 @@ public IEnumerable<BaseTemplate> Get()
...
+            var template = list.Where((temp) => temp.Name == templateName);
```
Comment (ahmelsayed) [inline SimpleWAWS/Controllers/TemplatesController.cs]: do we care about case insensitive check?

File: SimpleWAWS/Controllers/TemplatesController.cs [hunk 11]
```diff
@@ -39,5 +46,18 @@ public IEnumerable<BaseTemplate> Get()
...
+            if (template.Any())
+            {
+                var armTemplateJson = TemplatesManager.GetARMTemplate(template.FirstOrDefault());
```
Comment (ahmelsayed) [inline SimpleWAWS/Controllers/TemplatesController.cs]: If you just care about FirstOrDefault anyway, then why not just use it instead of Where above then check for null?

File: SimpleWAWS/ARMTemplates/BaseARMTemplate.json (hunk 1 above)
Comment (ahmelsayed) [inline SimpleWAWS/ARMTemplates/BaseARMTemplate.json]: it should be moved to a storage account and not on github.

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 12]
```diff
@@ -60,35 +64,72 @@ static TemplatesManager()
...
+        private static void updateParameters(ref dynamic temp, BaseTemplate template)
+        {
+            var shortName = getShortName(template.Name);
+            temp.parameters.appServiceName.defaultValue = string.Concat("My-", shortName, "App-", Guid.NewGuid().ToString().Split('-')[0]);
+            temp.parameters.msdeployPackageUrl.defaultValue = template.MSDeployPackageUrl;
+        }
+
+        private static void updateConfig(ref dynamic temp, BaseTemplate template)
+        {
+            temp.resources[1].resources[0].properties.templateName = template.Name ;
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: this seems to be to bake too much knowledge of a dynamic object structure in the code. isn't there a way to do this with static type validation?

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 13]
```diff
@@ -60,35 +64,72 @@ static TemplatesManager()
...
+        }
+
+        private static void updateAppSettings(ref dynamic temp, BaseTemplate template)
+        {
+            temp.resources[1].properties.siteConfig.appSettings.Add(new JObject {
+                { "name","foo"+ template.Name },
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: formatting, and use string interpolation rather than + concat

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 14]
```diff
@@ -60,35 +64,72 @@ static TemplatesManager()
...
+                { "value","bar" },
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: foo and bar? are these really needed or some testing leftovers?

File: SimpleWAWS/Controllers/TemplatesController.cs [hunk 15]
```diff
@@ -39,5 +46,18 @@ public IEnumerable<BaseTemplate> Get()
...
+                return Request.CreateResponse(HttpStatusCode.OK, armTemplateJson, new MediaTypeHeaderValue("application/json"));
+            }
+            else {
```
Comment (ahmelsayed) [inline SimpleWAWS/Controllers/TemplatesController.cs]: place { on a separate line

File: SimpleWAWS/Models/ResourceGroup.cs (hunk 8 above)
Comment (ahmelsayed) [inline SimpleWAWS/Models/ResourceGroup.cs]: use `string.IsNullOrEmpty()` instead

File: SimpleWAWS/Models/ResourceGroup.cs [hunk 16]
```diff
@@ -181,6 +181,8 @@ public UIResource UIResource
...
+                    templateName = "Ping Site";
```
Comment (ahmelsayed) [inline SimpleWAWS/Models/ResourceGroup.cs]: temp.resources[1].resources[0].properties.templateName

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 17]
```diff
@@ -60,35 +64,72 @@ static TemplatesManager()
...
+            updateParameters(ref armTemplate, template);
+            //TODO: Add mroe specific app settings as needed.
+            updateAppSettings(ref armTemplate, template);
+            updateConfig(ref armTemplate, template);
+            return armTemplate;
+        }
+        private static void updateParameters(ref dynamic temp, BaseTemplate template)
+        {
+            var shortName = getShortName(template.Name);
+            temp.parameters.appServiceName.defaultValue = string.Concat("My-", shortName, "App-", Guid.NewGuid().ToString().Split('-')[0]);
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: I'd use string interpolation or string.Format instead of string.Concat here. I think that would make it clearer.

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 18]
```diff
@@ -79,16 +85,50 @@ static TemplatesManager()
...
+        public static JObject GetARMTemplate(BaseTemplate template)
+        {
+            var armTemplate = JObject.FromObject(_baseARMTemplate);
+            updateParameters( armTemplate, template);
+            //TODO: Add more specific app settings as needed.
+            updateAppSettings(armTemplate, template);
+            updateConfig(armTemplate, template);
+            return armTemplate;
+        }
+        private static void updateParameters(dynamic temp, BaseTemplate template)
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: add an empty line before function declaration

Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: use PascalCasing for functionNames

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 19]
```diff
@@ -79,16 +85,50 @@ static TemplatesManager()
...
+        {
+            var shortName = GetShortName(template.Name);
+            temp.parameters.appServiceName.defaultValue = $"{Server.ARMTemplate_MyPrefix}-{shortName}{Server.ARMTemplate_AppPostfix}-{Guid.NewGuid().ToString().Split('-')[0]}";
+            temp.parameters.msdeployPackageUrl.defaultValue = template.MSDeployPackageUrl;
+        }
+
+        private static void updateConfig(dynamic temp, BaseTemplate template)
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: PascalCasing

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 20]
```diff
@@ -79,16 +85,50 @@ static TemplatesManager()
...
+        {
+            temp.resources[1].resources[0].properties.templateName = template.Name ;
+        }
+
+        private static void updateAppSettings(dynamic temp, BaseTemplate template)
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: PascalCasing

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 21]
```diff
@@ -79,16 +85,50 @@ static TemplatesManager()
...
+        {
+            temp.resources[1].properties.siteConfig.appSettings.Add(new JObject {
+                { "name","foo"+ template.Name },
+                { "value","bar" },
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: are these needed?

File: SimpleWAWS/Models/BaseTemplate.cs [hunk 22]
```diff
@@ -25,5 +25,7 @@ public string CreateQueryString()
         {
             return string.Concat("appServiceName=", AppService.ToString(), "&name=", Name, "&autoCreate=true");
         }
+        [JsonProperty(PropertyName = "msdeployPackageUrl")]
```
Comment (ahmelsayed) [inline SimpleWAWS/Models/BaseTemplate.cs]: empty line before a property.

File: SimpleWAWS/Models/ResourceGroup.cs [hunk 23]
```diff
@@ -181,8 +181,11 @@ public UIResource UIResource
                         break;
                 }
                 var templateName = Tags.ContainsKey(Constants.TemplateName) ? Tags[Constants.TemplateName] : string.Empty;
-
-                return (siteToUseForUi == null|| (AppService ==AppService.Jenkins))
+                if (string.IsNullOrEmpty(templateName) && AppService == AppService.Logic)
+                {
+                    templateName = TemplatesManager.GetTemplates().FirstOrDefault((template) => template.AppService == AppService.Logic).Name;
```
Comment (ahmelsayed) [inline SimpleWAWS/Models/ResourceGroup.cs]: ?.Name since you're using FirstOrDefault and that could return a null

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 24]
```diff
@@ -79,16 +85,50 @@ static TemplatesManager()
...
+        }
+
         public static IEnumerable<BaseTemplate> GetTemplates()
         {
             return _templatesList;
         }
+
+        public static JObject GetARMTemplate(BaseTemplate template)
+        {
+            var armTemplate = JObject.FromObject(_baseARMTemplate);
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: Can you add a comment explaining why you need to call this?

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 25]
```diff
@@ -79,16 +85,50 @@ static TemplatesManager()
...
+            updateParameters( armTemplate, template);
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: formatting (extra space)

File: SimpleWAWS/Code/TemplatesManager.cs [hunk 26]
```diff
@@ -60,17 +65,18 @@ static TemplatesManager()
...
                 list.Add(new JenkinsTemplate
                 {
                     Name = "Jenkins CI",
                     SpriteName = "sprite-PingSite PingSite",
                     AppService = AppService.Jenkins,
-                    CsmTemplateFilePath = HostingEnvironment.MapPath("~/CSMTemplates/JenkinsResource.json"),
+                    CsmTemplateFilePath = HostingEnvironment.MapPath("~/ARMTemplates/JenkinsResource.json"),
                     Description = Resources.Server.Templates_JenkinsDescription
                 });
+                _baseARMTemplate = JObject.Parse( File.ReadAllText(HostingEnvironment.MapPath("~/ARMTemplates/BaseARMTemplate.json")));
```
Comment (ahmelsayed) [inline SimpleWAWS/Code/TemplatesManager.cs]: Formatting.
Also can you add a comment explaining what this is and how it's used? It's not common to use JObject.FromObject to do a deep cloning of an object.

File: SimpleWAWS/ARMTemplates/BaseARMTemplate.json [hunk 27]
```diff
@@ -0,0 +1,103 @@
...
+        "serverFarmId": "[resourceId('Microsoft.Web/serverfarms/', variables('hostingPlanName'))]",
+        "siteConfig": {
+          "appSettings": [
+            {
+              "name": "WEBSITE_NODE_DEFAULT_VERSION",
+              "value": "6.5.0"
+            },
+            {
+              "Name": "SCM_SITEEXTENSIONS_FEED_URL",
+              "Value": "http://www.siteextensions.net/api/v2/"
```
Comment (ahmelsayed) [inline SimpleWAWS/ARMTemplates/BaseARMTemplate.json]: Why is this needed?

File: SimpleWAWS/ARMTemplates/BaseARMTemplate.json [hunk 28]
```diff
@@ -0,0 +1,103 @@
...
+        {
+          "apiVersion": "2015-08-01",
+          "name": "metadata",
+          "type": "config",
+          "dependsOn": [
+            "[resourceId('Microsoft.Web/Sites/', toLower(variables('appServiceName')))]",
+            "[resourceId('Microsoft.Web/Sites/Extensions/', toLower(variables('appServiceName')), 'MSDeploy')]"
+          ],
+          "properties": {
+            "synctriggersstatus": "[listsyncfunctiontriggerstatus(resourceId('Microsoft.Web/sites/', toLower(variables('appServiceName'))), '2015-08-01').status]"
```
Comment (ahmelsayed) [inline SimpleWAWS/ARMTemplates/BaseARMTemplate.json]: what is this for?

File: SimpleWAWS/ARMTemplates/BaseARMTemplate.json (hunk 1 above)
Comment (ahmelsayed) [inline SimpleWAWS/ARMTemplates/BaseARMTemplate.json]: Is this moving to storage?
